src/store/mmap.rs

```diff
@@ -239,13 +241,9 @@ impl<E: Element> Store<E> for MmapStore<E> {
     }
 
     fn compact(&mut self, _config: StoreConfig, _store_version: u32) -> Result<bool> {
-        if self.map.is_none() {
-            return Ok(false);
-        }
-
-        self.map = None;
+        let map = self.map.take();
 
-        return Ok(true);
+        Ok(map.is_some())
```

Comment:
Why not `Ok(self.map.is_none())` instead?

Reply:
because that would always return `false`, as I take the value out in the line above

Reply:
I guess I don't quite see the functional difference of:

```
let mut my_option = Some("hello");
let rest = my_option.take();
Ok(rest.is_some())

... and ...

my_option = None;
Ok(my_option.is_none())
```

Please excuse my rust newbness ;-)

Reply:
in the second part `my_option` gets set to `None`, and so the check `my_option.is_none()` will always be true, no matter what value `my_option` had before

Reply:
Ok, that was more or less what I intended, as it doesn't matter to me if it's called in succession, only that if it was there, it's dropped.  So I think the difference is that you want it to return an indication of error for that case, which makes sense too.  Thanks.